src/App.js

```diff
@@ -1,24 +1,36 @@
-import logo from './logo.svg';
 import './App.css';
+import { useEffect, useState } from 'react';
+import fetchData from './services/blog';
+import PostList from './components/PostList';
 
 function App() {
+  const [posts, setPosts] = useState([])
+  const [hasError, setHasError] = useState(false)
+  const [isLoading, setIsLoading] = useState()
+
+  const getPost = () => {
+    setIsLoading(true)
+
+    fetchData()
+      .then((newPosts) => {
+        setIsLoading(false)
+        setPosts(newPosts)
+      })
+      .catch((error) => {
+        setHasError(true)
```

Comment:
I think you also need to remove the loading state here, otherwise, the loading state will persist if the `fetchData` throws an error